.github/workflows/main.yml

```diff
@@ -0,0 +1,46 @@
+name: CMake
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        BUILD_SHARED_LIBS: ["ON", "OFF"]
+        ENABLE_DTLS: ["ON", "OFF"]
+        ENABLE_TCP: ["ON", "OFF"]
+        DTLS_BACKEND: ["openssl", "gnutls", "mbedtls"]
```

Comment:
current matrix check, we could add windows and mac to this file in the future.
tinyTLS is still missing here, could be added in the future